src/array/cuda/segment_reduce.cu

```diff
@@ -48,6 +50,18 @@ void ScatterAdd(NDArray feat,
 }
 
 
+template <int XPU, typename IdType, int bits>
+void UpdateGradMinMax_hetero(HeteroGraphPtr g,
+                       std::vector<NDArray> feat,
+                       std::vector<NDArray> idx,
+                       std::vector<NDArray> idx_etype,
+                       std::vector<NDArray> out) {
+  SWITCH_BITS(bits, DType, {
+    // cuda::ScatterAdd<IdType, DType>(feat, idx, out);
+  });
```

Comment:
Raise an error.

You can also as well not write this function and use `ATEN_XPU_SWITCH` instead of `ATEN_XPU_SWITCH_CUDA` in `ScatterAddDispatchHetero`.

Reply:
Raised an error.